tools/ports/boost_headers.py

```diff
@@ -55,10 +56,9 @@ def clear(ports, shared):
 
 
 def process_args(ports, args, settings, shared):
-  if settings.USE_BOOST_HEADERS == 1:
-    get(ports, settings, shared)
-    args += ['-DBOOST_ALL_NO_LIB']
-  return args
+  assert(needed(settings))
```

Comment:
how about removing the assert in each port? they look like excessive boilerplate. they are just called from a single place now, I think, which is safe?

(if we do want asserts, the outer `()` are not needed)